### PR TITLE
Add commands and CI for QA

### PR DESCRIPTION
**Pull request by koriym**
Added composer command and CI script for quality assurance.
Will this help for future release ?

# Composer Commands

* `composer test` for phpunit test
* `composer coverage` for test coverage
* `composer cs` for coding standard check
* `composer cs-fix` for coding standard fix
* `composer phpmd` for static analysis (1)
* `composer phpstan` for static analysis (2)
* `composer tests` for all tests above

# Travis CI

In addition to phpunit test, test as well as ...
* lowest dependencies
* coding standards
* static analysis by phpstan

### CODE REVIEW

**Comment by tanakahisateru** [comment]
PHP 7.2 + `--prefer-lowest`.

https://travis-ci.org/tanakahisateru/pinoco/jobs/385448151#L486
```
...................................PHP Fatal error:  Declaration of SebastianBergmann\Comparator\DOMNodeComparator::assertEquals($expected, $actual, $delta = 0, $canonicalize = false, $ignoreCase = false) must be compatible with SebastianBergmann\Comparator\ObjectComparator::assertEquals($expected, $actual, $delta = 0, $canonicalize = false, $ignoreCase = false, array &$processed = Array) in /home/travis/build/tanakahisateru/pinoco/vendor/sebastian/comparator/src/DOMNodeComparator.php on line 149
```

PHPUnit 4.8.x looks not compatible with PHP 7.2's new type checker but it has no constraint for upper version: `php >= 5.3.3`.

I think, `--prefer-lowest` case is not needed because Pinoco has no runtime dependencies. This project uses Composer for only `require-dev` at least between v0.x to v1.0.

**Comment by tanakahisateru** [comment]
I'm working to pick some ideas from this PR.

File: .travis.yml [hunk 1]
```diff
@@ -7,11 +8,20 @@ php:
...
 before_script:
   - composer self-update
   - composer update
-script: ./vendor/bin/phpunit
+  - if [[ $TRAVIS_PHP_VERSION = '7.1' ]]; then COVERAGE="--coverage-clover=coverage.clover"; else phpenv config-rm xdebug.ini; fi
+  - if [[ $TRAVIS_PHP_VERSION = '7.1' ]]; then composer require --dev phpstan/phpstan-shim friendsofphp/php-cs-fixer; fi
+script:
+  - ./vendor/bin/phpunit $COVERAGE;
+  - if [[ $TRAVIS_PHP_VERSION = '7.1' ]]; then IFS=$'\n'; COMMIT_SCA_FILES=($(git diff --name-only --diff-filter=ACMRTUXB "${TRAVIS_COMMIT_RANGE}")); unset IFS && ./vendor/bin/php-cs-fixer fix ./ --rules=@PSR2 -v --dry-run --using-cache=no -- "${COMMIT_SCA_FILES[@]}";fi
+  - if [[ $TRAVIS_PHP_VERSION = '7.1' ]]; then ./vendor/bin/phpstan analyse -l max -c phpstan.neon src test --no-progress --no-interaction; fi
```
Comment (tanakahisateru) [inline .travis.yml]: `[[$TRAVIS_PHP_VERSION == '7.1']]`

**Comment by tanakahisateru** [comment]
Sorry not to merge the PR, instead it was partially imported: https://github.com/tanakahisateru/pinoco/compare/d4cdcf0370b26decfda15d30919c9626b6df2efc...110e54a3fae3b09ad0ec6a030beaffdf84c5b649

**Comment by tanakahisateru** [comment]
Why not check types for tests?: Too many magic methods are tested there.
Why so much ignored naming rules?: The naming rules of Pinoco were designed before PSR-0.

**Comment by tanakahisateru** [comment]
@koriym Please check my new master:HEAD again 🙇

**Comment by koriym** [comment]
You can rebase my PR then I will know which part are taken and which part are dropped.

**Comment by koriym** [comment]
Oh you already commit to master directly, Then forget last comment. You commits seems be OK,
(But I feel it would be nice if yon do not drop test `tests/` `example/` though)

**Pull request closed without merging**